tests/unit/test-firefox/test.firefox.js

```diff
@@ -511,20 +511,136 @@ describe('firefox', () => {
          } catch (error) {
            exception = error;
          }
-
+         assert.instanceOf(exception, UsageError);
          assert.match(
            exception && exception.message,
            /Cannot use --keep-profile-changes on a default profile/
          );
        });
 
+    it('rejects to a UsageError when profle is not found',
+       async () => {
+         const fakeGetProfilePath = sinon.spy(() => Promise.resolve(false));
+         const createProfileFinder = () => {
+           return fakeGetProfilePath;
+         };
+         const isFirefoxDefaultProfile = sinon.spy(
+           () => Promise.resolve(false)
+         );
+         let exception;
+         try {
+           await firefox.useProfile('profileName', {
+             createProfileFinder,
+             isFirefoxDefaultProfile,
+           });
+         } catch (error) {
+           exception = error;
+         }
+         assert.instanceOf(exception, UsageError);
+         assert.match(
+           exception && exception.message,
+           /The request "profileName" profile name cannot be resolved/
+         );
+       }
+    );
+
     it('resolves to a FirefoxProfile instance', () => withBaseProfile(
-      (baseProfile) => {
-        const configureThisProfile = (profile) => Promise.resolve(profile);
-        return firefox.useProfile(baseProfile.path(), {configureThisProfile})
-          .then((profile) => {
-            assert.instanceOf(profile, FirefoxProfile);
+      async (baseProfile) => {
+        try {
+          const app = 'fennec';
+          const configureThisProfile = (profile) => Promise.resolve(profile);
+          const createProfileFinder = () => {
+            return (profilePath) => Promise.resolve(profilePath);
+          };
+          const profile = await firefox.useProfile(baseProfile.path(), {
+            app,
+            configureThisProfile,
+            createProfileFinder,
           });
+          assert.instanceOf(profile, FirefoxProfile);
+        } catch (error) {
```

Comment:
This `try...catch` is just re-throwing the caught error, it doesn't seem to be needed (same for the other ones in the other new tests below).